### DISCUSSION TITLE
Do i need space for the shared files on my server?

### DISCUSSION THREAD

**Discussion by Musabe24** [schlagmichdoch/PairDrop#344] [Q&A] [answered]
Hello. Thanks for this fork I love it. But I am asking how this works. For example I want to share a file with the size of 10 GB. Do I need 10 GB free space on my server for that?

**Comment by schlagmichdoch** [comment, marked as answer]
Sorry for the late reply! Somehow this got buried!

> I want to share a file with the size of 10 GB. Do I need 10 GB free space on my server for that?

No. All connections are duplex peer-to-peer connections: files are directly sent from receiver to the server. For internet transfers you need to deploy a TURN server alongside PairDrop (see [docs](https://github.com/schlagmichdoch/PairDrop/blob/master/docs/host-your-own.md#turn-server-for-internet-transfer)) to relay the traffic. For some connections, the encrypted transfer is then relayed by the server. No files are ever uploaded to the server. See https://github.com/schlagmichdoch/PairDrop/blob/master/docs/technical-documentation.md

**Reply by Musabe24** [reply]
Thanks!
